modules/test/playwright/pages/knowledge-base-web/KnowledgeBaseViewArticle.page.ts

```diff
@@ -11,13 +11,14 @@ export class KnowledgeBaseViewArticlePage {
 	deleteMenuItem: Locator;
 
 	constructor(page: Page) {
-		this.showActionsButton = page.getByLabel('Show Actions');
+		this.showActionsButton = page
+			.getByLabel('Show Actions')
+			.and(page.locator('[aria-haspopup]'));
 		this.page = page;
 	}
 
 	async deleteKnowledgeBaseArticle() {
-		// await this.showActionsButton.locator('[aria-haspopup]').waitFor();
-
+		await this.showActionsButton.waitFor();
```

Comment:
Would it be a good idea to put the `waitFor` inside the `constructor` so we know `showActionsButton` is already available when using it? Also, can the `waitFor` + second condition be replaced more clearly by an [`expect`](https://playwright.dev/docs/actionability#assertions) ?

Reply:
Nop, the constructor code must be sync

Reply:
the async deleteKnowledgeBaseArticle() {
is not a test, so an expect is... not expected
<img src="https://media4.giphy.com/media/Cw5MHCXkhk9YxhihZd/giphy.gif"/>